File: sistema_cpluplus/codigos/ItemFolhaDePonto.cpp
#include "ItemFolhaDePonto.h"
#include <iostream>//entrada e saida

using namespace std;

//construtor

ItemFolhaDePonto::ItemFolhaDePonto(string observacao)
{

	if (!setObservacao(observacao))
	{
		this->observacao = " ";
	}
}

//construtor

ItemFolhaDePonto::ItemFolhaDePonto(Data& data, string observacao)
{
	this->data = data;


	if (!setObservacao(observacao))
	{
		this->observacao = " ";
	}
}

//construtor
ItemFolhaDePonto::ItemFolhaDePonto(ItemFolhaDePonto& item)
{
	this->data = item.data;
	this->observacao = item.observacao;

}



//destrutor

ItemFolhaDePonto:: ~ItemFolhaDePonto() {}

//getters

string ItemFolhaDePonto::getObservacao()
{
	return observacao;
}

//setters

bool ItemFolhaDePonto::setObservacao(string observacao)
{

	if (observacao == "" || observacao == " ")//conferir se nome eh valido, validacao fraca mas ajuda um pouco
	{
		return false;
	}

	this->observacao = observacao;

	return true;
}

//sobrecarga do operador de atribuicao '=' 

ItemFolhaDePonto& ItemFolhaDePonto:: operator=(ItemFolhaDePonto& item)
{
	if (this == &item)
	{
		return *this;
	}

	this->data = item.data;
	this->observacao = item.observacao;

	return *this;
}

//sobrecarga do operador 'cout'
ostream& operator << (ostream& out, ItemFolhaDePonto& item)
{
	out << "Data: " << item.data << "  Observacao: " << item.observacao;
	return out;
}

File: sistema_java/codigos/oficinapoo/ItemFolhaDePonto.java

package oficinapoo;


public class ItemFolhaDePonto implements Comparable<ItemFolhaDePonto>
{
    public Data data;
    private String observacao;
    
    //construtores
    public ItemFolhaDePonto()
    {
        data = new Data();
        observacao = " "; 
    }
    
    public ItemFolhaDePonto(String observacao)
    {
        data = new Data();
        if (!setObservacao(observacao))
	{
            this.observacao = " ";
	}
    }
    
    
    public ItemFolhaDePonto(Data data, String observacao)
    {
        this.data = new Data(data);
         if (!setObservacao(observacao))
	{
            this.observacao = " ";
	}
    }

    public ItemFolhaDePonto(ItemFolhaDePonto item)
    {
       this.data = new Data(item.data);
       this.observacao = item.observacao;  
    }
    
    //getters
    public String getObservacao()
    {
        return observacao;
    }

    //setters
    public boolean setObservacao(String observacao)
    {
        if (observacao.equals("") || observacao.equals(" "))//conferir se nome eh valido, validacao fraca mas ajuda um pouco
	{
            return false;
	}

	this.observacao = observacao;

	return true;
    }
    
    //metodo comparavel
    public int compareTo(ItemFolhaDePonto item)
    {
        return this.data.compareTo(item.data);
    }
   
    //sobrecarga do operador 'printf'
    public String toString()
    {
        return String.format("Data: %s Observacao: %s",this.data,this.observacao);
    }

}
